Content.Shared/Xenoarchaeology/Artifact/XAT/Components/XATReactiveComponent.cs

```diff
@@ -0,0 +1,25 @@
+using Content.Shared.Chemistry;
+using Content.Shared.Chemistry.Reagent;
+using Content.Shared.FixedPoint;
+using Robust.Shared.GameStates;
+using Robust.Shared.Prototypes;
+
+namespace Content.Shared.Xenoarchaeology.Artifact.XAT.Components;
+
+/// <summary>
+/// This is used for a xenoarch trigger that activates when a reaction occurs on the artifact.
+/// </summary>
+[RegisterComponent, NetworkedComponent, Access(typeof(XATReactiveSystem)), AutoGenerateComponentState]
+public sealed partial class XATReactiveComponent : Component
+{
+    [DataField, AutoNetworkedField]
+    public List<ReactionMethod> ReactionMethods = new() { ReactionMethod.Touch };
+
+    [DataField, AutoNetworkedField]
+    public HashSet<ProtoId<ReagentPrototype>> Reagents = new();
+
+    //todo: ReactiveGroupPrototype
+
+    [DataField, AutoNetworkedField]
```

Comment:
my reactive group code is kinda questionable - way that Emo used to handle reactive events is by checking events of each reagent in the mix separately, so when my code checks for reactive group - it checks for amount of only 1 reagent at a time. so if you have acidic 5u of solution, with 1u of reagents, code will see only 1u at a time. I think it should be fixed but it is kinda separate story - we better off remove ReativeGroup from here in this PR at all, or leave it as it is.